ffmpeg/dag.py

```diff
@@ -69,31 +70,38 @@ def incoming_edge_map(self):
         raise NotImplementedError()
 
 
-DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label'])
+DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label', 'upstream_selector'])
 
 
 def get_incoming_edges(downstream_node, incoming_edge_map):
     edges = []
-    for downstream_label, (upstream_node, upstream_label) in list(incoming_edge_map.items()):
-        edges += [DagEdge(downstream_node, downstream_label, upstream_node, upstream_label)]
+    # downstream_label, (upstream_node, upstream_label) in [(i[0], i[1][:2]) for i in self.incoming_edge_map.items()]
+    for downstream_label, upstream_info in [(i[0], i[1]) for i in incoming_edge_map.items()]:
+        upstream_node, upstream_label = upstream_info[:2]
+        upstream_selector = None if len(upstream_info) < 3 else upstream_info[2]
+        edges += [DagEdge(downstream_node, downstream_label, upstream_node, upstream_label, upstream_selector)]
     return edges
 
 
 def get_outgoing_edges(upstream_node, outgoing_edge_map):
     edges = []
     for upstream_label, downstream_infos in list(outgoing_edge_map.items()):
-        for (downstream_node, downstream_label) in downstream_infos:
-            edges += [DagEdge(downstream_node, downstream_label, upstream_node, upstream_label)]
+        for downstream_info in downstream_infos:
+            downstream_node, downstream_label = downstream_info[:2]
```

Comment:
What's `[:2]`?

Reply:
I think this would do the same thing (without the mysterious `:2`):
```
-            downstream_node, downstream_label = downstream_info[:2]
+            downstream_node, downstream_label = downstream_info
```

Reply:
Done 14b64706